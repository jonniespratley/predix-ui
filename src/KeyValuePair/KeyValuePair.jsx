import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from '../styled';


/*

.alpha { font-size:5rem;line-height:1.06667;font-weight:400;}
.beta { font-size: 4rem; line-height: 1; font-weight: 400; }
.delta, .gamma { line-height: 1.33333; font-weight: 400; }
.gamma { font-size: 3rem; }
.delta { font-size: 2rem; }
.epsilon { font-size: 1.33333rem; line-height: 1; font-weight: 400; }
.zeta { font-size: .8rem; line-height: 1.66667; font-weight: 400; }
 */

const fontSizes = {
  alpha: '5rem',
  beta: '4rem',
  delta: '2rem',
  gamma: '3rem',
  epsilon: '1.3333rem',
  zeta: '.8rem'
};

/* eslint-disable */
const _getAdjustedSize = s => (
  s === 'alpha' ? 'delta' :
    s === 'beta' ? 'epsilon' :
      s === 'gamma' ? 'value' :
        s === 'delta' ? 'value' : 'zeta'
);
/* eslint-enable */

const KvpLabel = styled.div`
  font-size: .8rem;
  line-height: 1.66667;
  color: var(--px-headings-label-color,gray);
`;
KvpLabel.displayName = 'KvpLabel';
KvpLabel.defaultProps = {
  className: 'kvp-label',
  size: null
};
KvpLabel.propTypes = {
  className: PropTypes.string,
  size: PropTypes.string
};

const KvpValue = styled.span`
  color: var(--px-headings-value-color, black);
  margin-top: -4px;
  ${props => props.size && css`
    font-size: ${fontSizes[props.size]};
  `}
  ${props => props.size === 'alpha' && css`
    margin-top: -10px;
  `}
  ${props => props.size === 'beta' && css`
    margin-top: -6px;
  `}
  ${props => props.size === 'gamma' && css`
    margin-top:-12px;
  `}
  ${props => props.size === 'delta' && css`
    margin-top: -8px;
  `}
  ${props => props.size === 'epsilon' && css`
    margin-top: -2px;
  `}
`;
KvpValue.displayName = 'KvpValue';
KvpValue.defaultProps = {
  className: 'kvp-value',
  size: null
};
KvpValue.propTypes = {
  className: PropTypes.string,
  size: PropTypes.string
};

const KvpUom = styled.span`
  margin-left: 0px;
  
  ${props => props.size && css`
    font-size: ${fontSizes[props.size]};
  `}
  ${props => props.size === 'value' && css`
    font-size: 1rem;
  `}
  ${props => props.size === 'alpha' && css`
    margin-left: -12px;
  `}
  ${props => props.size === 'beta' && css`
    margin-left: -10px;
  `}
  ${props => props.size === 'gamma' && css`
    margin-left: -8px;
  `}
  ${props => props.size === 'delta' && css`
    margin-left: -4px;
  `}
  ${props => props.size === 'epsilon' && css`
    margin-left: -2px;
  `}
`;
KvpUom.displayName = 'KvpUom';
KvpUom.defaultProps = {
  className: 'kvp-uom',
  size: null
};
KvpUom.propTypes = {
  className: PropTypes.string,
  size: PropTypes.string
};


const Component = ({
  label,
  value,
  size,
  uom,
  children
}) => (
  <div className="px-key-value-pair">
    <KvpLabel>{label}</KvpLabel>
    <KvpValue size={_getAdjustedSize(size)} className={size}>
      {value}
      {uom && (
      <KvpUom
        size={_getAdjustedSize(size)}
        className={_getAdjustedSize(size)}
      >
        {uom}
      </KvpUom>
      )}
    </KvpValue>
    {children}
  </div>
);

Component.defaultProps = {
  label: null,
  value: null,
  size: null,
  uom: null,
  children: null
};

Component.propTypes = {
  label: PropTypes.string,
  value: PropTypes.string,
  size: PropTypes.string,
  uom: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func])
};

export default Component;
