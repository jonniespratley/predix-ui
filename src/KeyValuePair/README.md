The `<KeyValuePair/>` component allows you to prominently display information in a dashboard.


## Usage

```react
span: 6
---
const { KeyValuePair } = PxReact;
const sizes = ['alpha', 'beta', 'gamma', 'delta', 'epsilon', 'regular'];
<div>
  {sizes.map((size, index) => (
    <KeyValuePair
      key={index}
      label="Lorem Ipsum"
      value="12345"
      uom="units"
      size={size}/>)
  )}
</div>
```


## Properties

```table
span: 6
rows:
  - Name: label (key)
    Type: String
    Description: The label (key) of the key-value pair

  - Name: size
    Type: String
    Description: The display size of the value in the key-value pair. (alpha, beta, gamma, delta, epsilon, regular)

  - Name: uom
    Type: String
    Description: The unit of measure for the value

  - Name: value
    Type: String
    Description: The value of the key-value pair
```
