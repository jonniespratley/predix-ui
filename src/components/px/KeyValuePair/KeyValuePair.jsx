import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const KvpLabel = styled.div`
  font-size: .8rem;
  line-height: 1.66667;
  color: var(--px-headings-label-color,gray);
`;
KvpLabel.displayName = 'KvpLabel';
KvpLabel.defaultProps = {
  className: 'kvp-label',
  size: null
};
KvpLabel.propTypes = {
  className: PropTypes.string,
  size: PropTypes.string
};

const KvpValue = styled.span`
  color: var(--px-headings-value-color, black);
  margin-top: -4px;
  ${props => props.size === 'alpha' && css`
    margin-top: -10px;
  `}
  ${props => props.size === 'beta' && css`
    margin-top: -6px;
  `}
  ${props => props.size === 'gamma' && css`
    margin-top:-12px;
  `}
  ${props => props.size === 'delta' && css`
    margin-top: -8px;
  `}
  ${props => props.size === 'epsilon' && css`
    margin-top: -2px;
  `}
`;
KvpValue.displayName = 'KvpValue';
KvpValue.defaultProps = {
  className: 'kvp-value',
  size: null
};
KvpValue.propTypes = {
  className: PropTypes.string,
  size: PropTypes.string
};

const KvpUom = styled.span`
  margin-left: 0px;
  ${props => props.size === 'value' && css`
    font-size: 1rem;
  `}
  ${props => props.size === 'alpha' && css`
    margin-left: -12px;
  `}
  ${props => props.size === 'beta' && css`
    margin-left: -10px;
  `}
  ${props => props.size === 'gamma' && css`
    margin-left: -8px;
  `}
  ${props => props.size === 'delta' && css`
    margin-left: -4px;
  `}
  ${props => props.size === 'epsilon' && css`
    margin-left: -2px;
  `}
`;
KvpUom.displayName = 'KvpUom';
KvpUom.defaultProps = {
  className: 'kvp-uom',
  size: null
};
KvpUom.propTypes = {
  className: PropTypes.string,
  size: PropTypes.string
};

/* eslint-disable */
const _getAdjustedSize = s => (
  s === 'alpha' ? 'delta' :
    s === 'beta' ? 'epsilon' :
      s === 'gamma' ? 'value' :
        s === 'delta' ? 'value' : 'zeta'
);
/* eslint-enable */

const Component = ({
  label,
  value,
  size,
  uom,
  children
}) => (
  <div className="px-key-value-pair">
    <KvpLabel >{label}</KvpLabel>
    <KvpValue size={_getAdjustedSize(size)} className={size}>
      {value}
      {uom &&
      <KvpUom size={_getAdjustedSize(size)} className={_getAdjustedSize(size)}>{uom}</KvpUom>}
    </KvpValue>
    {children}
  </div>
);

Component.defaultProps = {
  label: null,
  value: null,
  size: null,
  uom: null,
  children: null
};

Component.propTypes = {
  label: PropTypes.string,
  value: PropTypes.string,
  size: PropTypes.string,
  uom: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func])
};

export default Component;
