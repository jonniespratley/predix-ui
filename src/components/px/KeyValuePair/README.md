The `<KeyValuePair/>` component allows you to prominently display information in a dashboard.


## Usage

```react
span: 6
---
const { KeyValuePair } = PxReact;
<KeyValuePair
  label="Lorem Ipsum"
  value="12345"
  uom="units"
  size="alpha"/>
```


## Properties

```table
span: 6
rows:
  - Name: label (key)
    Type: String
    Description: The label (key) of the key-value pair

  - Name: size
    Type: String
    Description: The display size of the value in the key-value pair. (alpha, beta, gamma, delta, epsilon, regular)

  - Name: uom
    Type: String
    Description: The unit of measure for the value

  - Name: value
    Type: String
    Description: The value of the key-value pair
```
