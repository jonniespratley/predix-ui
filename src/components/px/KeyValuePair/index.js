import React from 'react';
import classnames from 'classnames';

//import stylesheet from './px-key-value-pair.scss';

/**
 * KeyValuePair component
 */
export default ({
  label = 'KeyValuePair',
  value,
  size,
  uom,
  children
}) => {

const _getAdjustedSize = (s) => {
  return (
    s === 'alpha' ? 'delta' :
    s === 'beta' ? 'epsilon' :
    s === 'gamma' ? 'value' :
    s === 'delta' ? 'value' : 'zeta'
  );
};
  const baseClassnames = classnames(
    'px-key-value-pair'
  );
  const valueClassnames = classnames(
    size,
    `kvp-value--${size}`
  );

  return (
    <div className={baseClassnames}>
      <div className='label'>{label}</div>
      <div className={valueClassnames}>
        {value}
        {uom && <span className={_getAdjustedSize(size)}>{uom}</span>}
      </div>
      
    </div>
  );
}
