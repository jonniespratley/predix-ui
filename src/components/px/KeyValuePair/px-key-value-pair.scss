/**
 * px-key-value-pair styles
 */
.px-key-value-pair {

  // Settings

  // Tools

  // Generic
  //@import "px-normalize-design/_generic.normalize.scss";
  //@import "px-box-sizing-design/_generic.box-sizing.scss";
  @import "px-helpers-design/_generic.helpers.scss";

  // Base
  @import "px-headings-design/_base.headings.scss";

  // Meta

  // Objects

  // Component

  .kvp-value {
    color: var(--px-headings-value-color, black);
  }

  .kvp-value--alpha {
    margin-top: calculateRem(-10px);
  }

  .kvp-value--beta {
    margin-top: calculateRem(-6px);
  }

  .kvp-value--gamma {
    margin-top: calculateRem(-12px);
  }

  .kvp-value--delta {
    margin-top: calculateRem(-8px);
  }

  .kvp-value--epsilon {
    margin-top: calculateRem(-2px);
  }

  .kvp-value--regular {
    margin-top: calculateRem(-4px);
  }

  
}
