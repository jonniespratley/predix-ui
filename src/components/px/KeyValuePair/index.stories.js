import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, text, array, object, boolean, number, select } from '@storybook/addon-knobs';


import KeyValuePair from './';


const sizes = ['alpha', 'beta', 'gamma', 'delta', 'epsilon', 'regular'];

storiesOf('KeyValuePair', module)
  .addDecorator(withKnobs)
  .add('default', () => (
    <KeyValuePair
      key={text('key', 'item-1')}
      label={text('label', 'Lorem Ipsum')}
      value={text('value', '12345')}
      uom={text('uom', 'units')}
      size={select('size', sizes, 'beta')}
    />
  ));
