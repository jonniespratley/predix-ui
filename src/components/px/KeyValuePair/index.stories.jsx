import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, select } from '@storybook/addon-knobs';
import KeyValuePair from './';

const sizes = ['alpha', 'beta', 'gamma', 'delta', 'epsilon', 'regular'];

storiesOf('Components / KeyValuePair', module)
  .addDecorator(withKnobs)
  .add('default', () => (
    <KeyValuePair
      key={text('key', 'item-1')}
      label={text('label', 'Lorem Ipsum')}
      value={text('value', '12345')}
      uom={text('uom', 'units')}
      size={select('size', sizes, 'beta')}
    />
  ));
