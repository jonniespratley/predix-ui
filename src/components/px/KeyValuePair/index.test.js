import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import KeyValuePair from './';

const sizes = ['alpha', 'beta', 'gamma', 'delta', 'epsilon', 'regular'];

describe('KeyValuePair', () => {
  test(`beta - renders correctly`, () => {
    const tree = renderer.create(
      <KeyValuePair label="Lorem Ipsum" value="12345" uom="units" size="beta"/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test(`alpha - renders correctly`, () => {
    const tree = renderer.create(
      <KeyValuePair label="Lorem Ipsum" value="12345" uom="units" size="alpha"/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  sizes.forEach(size => {
    test(`should render size ${size}`, () =>{
      const wrapper = shallow(
        <KeyValuePair
          label="Lorem Ipsum"
          value="12345"
          uom="units"
          size={size}/>
      );
      expect(wrapper.find(`.${size}`)).toHaveLength(1);
      expect(wrapper.find(`KvpValue`)).toHaveLength(1);
      expect(wrapper.find(`KvpValue`).name()).toEqual('KvpValue');
    });
  });
});
