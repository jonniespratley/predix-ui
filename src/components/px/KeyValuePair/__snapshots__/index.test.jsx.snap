// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyValuePair alpha - renders correctly 1`] = `
<div
  className="px-key-value-pair"
>
  <div
    className="kvp-label ctwgUQ"
    size={null}
  >
    Lorem Ipsum
  </div>
  <span
    className="alpha kfFMcF"
    size="delta"
  >
    12345
    <span
      className="delta eHtIjR"
      size="delta"
    >
      units
    </span>
  </span>
</div>
`;

exports[`KeyValuePair beta - renders correctly 1`] = `
<div
  className="px-key-value-pair"
>
  <div
    className="kvp-label ctwgUQ"
    size={null}
  >
    Lorem Ipsum
  </div>
  <span
    className="beta dEiQCL"
    size="epsilon"
  >
    12345
    <span
      className="epsilon bYwwOz"
      size="epsilon"
    >
      units
    </span>
  </span>
</div>
`;
